<a href="https://colab.research.google.com/github/visiont3lab/machine-learning-fav/blob/master/notebook/Introduzione_generale.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Introduzione




In [ ]:
from google.colab import drive
drive.mount('/content/gdrive')

Go to this URL in a browser: https://accounts.google.com/o/oauth2/auth?client_id=947318989803-6bn6qk8qdgf4n4g3pfee6491hc0brc4i.apps.googleusercontent.com&redirect_uri=urn%3aietf%3awg%3aoauth%3a2.0%3aoob&response_type=code&scope=email%20https%3a%2f%2fwww.googleapis.com%2fauth%2fdocs.test%20https%3a%2f%2fwww.googleapis.com%2fauth%2fdrive%20https%3a%2f%2fwww.googleapis.com%2fauth%2fdrive.photos.readonly%20https%3a%2f%2fwww.googleapis.com%2fauth%2fpeopleapi.readonly

Enter your authorization code:
··········
Mounted at /content/gdrive


In [ ]:
%cd /content/gdrive/My Drive/Colab Notebooks/machine_learning_fav

/content/gdrive/My Drive/Colab Notebooks/machine_learning_fav


In [ ]:
!pwd

/content/gdrive/My Drive/Colab Notebooks/machine_learning_fav


In [ ]:
import pandas as pd

df = pd.read_excel("dataset/data_barret_new_0.xlsx")
df

A-CONSTANT  k steep  k flat   k av   ACD     AXL  Target
0       118.98     43.1    42.5  42.80  2.61  23.060    0.00
1       118.98     44.2    43.7  43.95  3.61  25.697   -2.75
2       118.98     43.8    43.4  43.60  3.23  23.639    0.00
3       118.98     45.1    44.6  44.85  3.09  23.349    0.00
4       118.98     44.6    44.0  44.30  3.04  22.932    0.00
..         ...      ...     ...    ...   ...     ...     ...
86      118.98     43.6    43.1  43.35  3.57  25.783   -3.00
87      118.98     44.0    43.4  43.70  3.45  28.013   -2.50
88      118.98     42.4    41.1  41.75  3.59  26.336    0.00
89      118.98     47.2    46.5  46.85  3.54  22.591    0.00
90      118.98     43.2    42.5  42.85  3.29  23.949    0.00

[91 rows x 7 columns]

HO CAMBIATO QUALCOSA

## Informazioni  

### Docenti

* Manuel Rucci    manuel.rucci@t3lab.it
* Daniele Grotti  danielegrottiuk@gmail.com

### Domande

* Per domande relative ad argomenti fatti a lezione o dubbi di qualunque tipo come installazione di programmi o errori di codice scrivere a questa email visiont3lab@gmail.com

### Materiale del Corso

* [Github Repository](https://github.com/visiont3lab/tecnologie_data_science)
* [Website Book](https://visiont3lab.github.io/tecnologie_data_science/)

#### Extra

Materiale associato ai corsi di data visualization e machine learning.
* [Materiale del corso Data Visualization](https://github.com/visiont3lab/data-visualization)
* [Materiale del corso Machine Learning](https://github.com/visiont3lab/machine_learning)




## Linguaggio di Programmazione: Python

[Pyhton](https://www.python.org/) è un linguaggio di programmazione interpretato di alto livello. Esso fu create da Guido an Rossum nel 1991. Sono state rilasciate diverse versioni di Python in particolare Python2 e Python3 e in questo corso utilizzermo Python3 in quanto Python2 è deprecato. La scelta di utilizzare python come linguaggio è dovuta al fatto che esso offre una grande varietà di pacchetti (o librerie) che permettoro facilmente di approcciare al mondo di data-science. Utilizzando Python è possibile visualizzare grafici 2D e 3D utilizzando librerie come matplotlib, seaborn, echarts, plotly, creare algoritmi di Regressisione  e Classificazione per analizzare dati usando librerie come scikit-learn, utilizzare il calcolo vettoriale e matriciale con numpy e anche manipolare tabelle di dati (come fogli excel) usando pandas.


## Ambiente di Sviluppo Online: Google-Colaboratory


Utilizzermo [Google-Colaboratory](https://colab.research.google.com/notebooks/welcome.ipynb) come piattorma di scrittura ed esecuzione codice. Quest'ultimo è un ambiente di sviluppo creato dal Google Research Team che permette a chiunque gratuitamente di eseguire codice nel browser. In particolare esso fornisce all'utente la possibilitá di editare Jupyter Notebooks senza che l'utente debba installare alcun programma sul propio pc. Inoltre è connesso a google drive e per questo motivo ogni notebook editato attraverso questa piattaforma viene automaticamente salvato nel Google drive dell'utente. Questa piattorma è gratuita e offre risorse di computazione (12GB di Ram, High Performace CPU e Gpu) valide che permettono a chiunque di avere tutti i mezzi necessari per programmare e sviluppare nuovi algoritmi di machine learning, deep learning.  

In termini più tecnici  Gogole Colab è un servizio fornito da Google  a cui accediamo tramite browser. In particolare tramite il browser accediamo a un Jupyter notebook che sta andando su una macchina di Google che non richiede nessun setup per essere utilizzato, inoltre esso fornisce acesso gratutio a risorse computazionali elevate come GPU e TPU.

Per avere più informazioni sulle prinpali funzionalitá di Google Colaboratory consultare [Google Colab Welcome Notebook](https://colab.research.google.com/notebooks/welcome.ipynb)

## Ambiente di Sviluppo Offline: Anaconda 

[Anaconda Indivual Edition](https://www.anaconda.com/products/individual) è un insieme di tools tra cui librerie e ambiente di sviluppo pensato per data science.  Esso necessita di essere installato sul proprio pc ma allo stesso tempo l'installazione ha l'obbiettivo di fornire all'utente tutto il materiale necessario per approcciare al mondo di data science usando i linguaggi programmazione R e Python. Pertanto è un ottima scelta per avere un setup di data science installato sul proprio pc.

Per quanto riguarda l'installazione è consigliato scegliere **64-Bit Graphical Installer Python 3.7** poichè la versione Pyhton 2.7 è deprecata.
